### DISCUSSION TITLE
Specification (and extension) design guidelines

### DISCUSSION THREAD

**Discussion by arno-di-loreto** [OAI/sig-moonwalk#36] [General]
Just like API guidelines can facilitate the job of people creating API, having specification design guidelines could help people design OpenAPI v4, its extensions, its evolution, and future versions.
Creating guidelines is also an excellent way to find explanations for the rules we may (inadvertently) invent and especially evaluate the consequences for humans and machines reading/creating OpenAPI documents.

Here are some questions that should be answered in "OpenAPI Specification guidelines":
- How to point/reference another element? (#26) 
- Choosing between map and list (#32 and #25 ) 
- Naming conventions (#34) 

These are practical questions, guidelines can also describe more "meta" concerns like:
- Which perspectives/considerations should be evaluated when making a design decision?
- What is considered a breaking change?

**Comment by handrews** [comment]
The referencing discussion should keep the [analogous discussion at AsyncAPI](https://github.com/orgs/asyncapi/discussions/485) in mind.
